Source/WebCore/platform/graphics/cg/IOSurfacePool.cpp

```diff
@@ -76,9 +76,10 @@ Ref<IOSurfacePool> IOSurfacePool::create()
     return adoptRef(*new IOSurfacePool);
 }
 
-static bool surfaceMatchesParameters(IOSurface& surface, IntSize requestedSize, const DestinationColorSpace& colorSpace, IOSurface::Format format)
+static bool surfaceMatchesParameters(IOSurface& surface, IntSize requestedSize, const DestinationColorSpace& colorSpace, IOSurface::Format format, UseLosslessCompression useLosslessCompression)
 {
-    if (!surface.hasFormat(format))
+    // FIXME: It might be OK to take a surface that doesn't use compression when requesting one that does, but not the other way around.
```

Comment:
Probably not a great plan, though, because it could mean your pool gets full of un-compressed surfaces and you don't make compressed ones.